Issues

Hello, 

I'm running into the following error with the driver and ROS2 Galactic:

```
ros2 launch flir_spinnaker_ros2 chameleon.launch.py
[INFO] [launch]: All log files can be found below /home/rst/.ros/log/2022-04-22-16-00-24-523416-rst-leaf-588698
[INFO] [launch]: Default logging verbosity is set to INFO
[<launch.substitutions.launch_configuration.LaunchConfiguration object at 0x7f7a1ccdcc40>, '_']
[INFO] [camera_driver_node-1]: process started with pid [588700]
[camera_driver_node-1] [INFO] [1650636024.730163265] [chameleon_s]: debug: 0
[camera_driver_node-1] [INFO] [1650636024.730450522] [chameleon_s]:  serial: 18497292
[camera_driver_node-1] [INFO] [1650636024.736319797] [chameleon_s]: using default calibration URL
[camera_driver_node-1] [INFO] [1650636024.736406129] [chameleon_s]: camera calibration URL: file:///home/rst/.ros/camera_info/chameleon_s.yaml
[camera_driver_node-1] [ERROR] [1650636024.736527862] [camera_calibration_parsers]: Unable to open camera calibration file [/home/rst/.ros/camera_info/chameleon_s.yaml]
[camera_driver_node-1] [WARN] [1650636024.736546182] [chameleon_s]: Camera calibration file /home/rst/.ros/camera_info/chameleon_s.yaml not found
[camera_driver_node-1] [INFO] [1650636026.068137220] [chameleon_s]: using spinnaker lib version: 2.3.0.77
[camera_driver_node-1] [WARN] [1650636026.171879187] [chameleon_s]: overwriting bad param with default: parameter 'gain_auto' has invalid type: cannot declare a statically typed parameter with an uninitialized value
[camera_driver_node-1] terminate called after throwing an instance of 'rclcpp::exceptions::InvalidParameterTypeException'
[camera_driver_node-1]   what():  parameter 'gain_auto' has invalid type: cannot declare a statically typed parameter with an uninitialized value
[ERROR] [camera_driver_node-1]: process has died [pid 588700, exit code -6, cmd '/home/rst/autoware/install/flir_spinnaker_ros2/lib/flir_spinnaker_ros2/camera_driver_node --ros-args -r __node:=chameleon_s --params-file /tmp/launch_params_c32vf40m --params-file /tmp/launch_params_3v2czpi0 -r ~/control:=/exposure_control/control']
```


The issue seems to be related to [this change](https://github.com/ros2/rclcpp/blob/master/rclcpp/doc/notes_on_statically_typed_parameters.md) in ROS Galactic. The node runs, if I enable `dynamic_typing` for the `ni.descriptor` in the `createCameraParameters()` function. However, this would be just a quick fix.

Given that it is mentioned that this package runs on ROS Galactic, is there a fix known?

--- a/include/flir_spinnaker_ros2/camera_driver.h
+++ b/include/flir_spinnaker_ros2/camera_driver.h
@@ -86,7 +86,6 @@ private:
   double exposureTime_;  // in microseconds
   bool autoExposure_;    // if auto exposure is on/off
   bool dumpNodeMap_{false};
-  bool alwaysPublish_{false};
   bool debug_{false};
   bool computeBrightness_{false};
   uint32_t currentExposureTime_{0};

--- a/launch/blackfly_s.launch.py
+++ b/launch/blackfly_s.launch.py
@@ -11,7 +11,6 @@ camera_params = {
     # set parameters defined in blackfly_s.cfg    
     'gain_auto': 'Continuous',
     'exposure_auto': 'Continuous',
-    'always_publish': True,
     'frame_rate_auto': 'Off',
     'frame_rate': 20.0,
     'frame_rate_enable': True,

--- a/src/camera_driver.cpp
+++ b/src/camera_driver.cpp
@@ -184,7 +184,6 @@ void CameraDriver::readParameters()
   LOG_INFO("debug: " << debug_);
   cameraInfoURL_ = this->declare_parameter<std::string>("camerainfo_url", "");
   frameId_ = this->declare_parameter<std::string>("frame_id", get_name());
-  alwaysPublish_ = this->declare_parameter<bool>("always_publish", false);
   dumpNodeMap_ = this->declare_parameter<bool>("dump_node_map", false);
   qosDepth_ = this->declare_parameter<int>("image_queue_size", 4);
   computeBrightness_ =
@@ -469,7 +468,7 @@ void CameraDriver::doPublish(const ImageConstPtr & im)
 
   const std::string encoding = flir_to_ros_encoding(im->pixelFormat_);
 
-  if (pub_.getNumSubscribers() > 0 || alwaysPublish_) {
+  if (count_subscribers(pub_.getTopic()) > 0) {
     sensor_msgs::msg::CameraInfo::UniquePtr cinfo(
       new sensor_msgs::msg::CameraInfo(cameraInfoMsg_));
     // will make deep copy. Do we need to? Probably...
@@ -487,7 +486,7 @@ void CameraDriver::doPublish(const ImageConstPtr & im)
       publishedCount_++;
     }
   }
-  if (metaPub_->get_subscription_count() != 0 || alwaysPublish_) {
+  if (metaPub_->get_subscription_count() != 0) {
     metaMsg_.header.stamp = t;
     metaMsg_.brightness = im->brightness_;
     metaMsg_.exposure_time = im->exposureTime_;

